app/src/org/commcare/network/SimpleGetRequest.java

```diff
@@ -0,0 +1,88 @@
+package org.commcare.network;
+
+import org.commcare.logging.AndroidLogger;
+import org.commcare.utils.GlobalConstants;
+import org.javarosa.core.services.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.PasswordAuthentication;
+import java.net.URL;
+
+/**
+ * Makes simple redirect-following GET requests that can be authenticated or not
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+class SimpleGetRequest {
+
+    protected static InputStream makeRequest(final String username,
+                                          final String password,
+                                          URL url) throws IOException {
+        if (username == null || password == null) {
+            // clear any prior set authenticator to make unauthed requests
+            Authenticator.setDefault(null);
+        } else {
+            // make authenticated requests
+            Authenticator.setDefault(new Authenticator() {
+                @Override
+                protected PasswordAuthentication getPasswordAuthentication() {
+                    return new PasswordAuthentication(
+                            HttpRequestGenerator.buildDomainUser(username),
+                            password.toCharArray());
+                }
+            });
+        }
+        return makeModernRequest(url);
```

Comment:
I agree that this is convoluted code.
I was just lifting code unchanged out of `HttpRequestGenerator` into this file, so I didn't modify anything here. The eventual goal is to stop using apache http code completely, which is deprecated. So I'd like to leave cleaning this up until we get to that point.